docs/guides/themeable-to-emotion-migration-guide.md

```diff
@@ -0,0 +1,459 @@
+
+Merge the default variable object and `themeSpecificStyle[theme.key]` in the return block with spread operator, with the latter at the end, so it can override the defaults.
+
+**Before:**
+
+```js
+// ui-themeable
+
+export default function generator({ colors, borders, typography }) {
+  return {
+    color: colors.textBrand,
+    background: colors.backgroundLightest,
+    borderWidthSmall: borders.widthSmall,
+    borderWidthMedium: borders.widthMedium,
+    borderColor: colors.borderMedium,
+    fontFamily: typography.fontFamily,
+    fontWeight: typography.fontWeightBold
+  }
+}
+
+generator.canvas = function () {
+  return {
+    primaryColor: 'blue'
+  }
+}
+
+generator['canvas-high-contrast'] = function () {
+  return {
+    primaryColor: 'red'
+  }
+}
+```
+
+**After:**
+
+```js
+// emotion
+
+/**
+ * Generates the theme object for the component from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme) => {
+  // the props of the theme you want to use
+  const { colors, borders, typography, key: themeName } = theme
+
+  // if you need different styling in specific themes,
+  // add them here with the key of the theme
+  const themeSpecificStyle = {
+    'canvas-high-contrast': {
+      primaryColor: 'red'
+    },
+    canvas: {
+      primaryColor: 'blue'
+    }
+  }
+
+  // map the theme variables to component specific style variables,
+  // and use optional chaining (?.)
+  const componentVariables = {
+    color: colors?.textBrand,
+    background: colors?.backgroundLightest,
+    borderWidthSmall: borders?.widthSmall,
+    borderWidthMedium: borders?.widthMedium,
+    borderColor: colors?.borderMedium,
+    fontFamily: typography?.fontFamily,
+    fontWeight: typography?.fontWeightBold
+  }
+
+  // return with the final theme object of the component
+  // with the added theme specific overrides
+  return {
+    ...componentVariables,
+    ...themeSpecificStyle[themeName]
+  }
+}
+export default generateComponentTheme
+```
+
+#### 2. Create the styles.js file
+
+The two main advantages of the new approach are that
+
+- we can directly use the component theme variables in the styling _(instead of generating tons of CSS custom properties)_,
+- and we can pass the component props to the generator, so we can handle all prop-based styling here _(instead of adding multiple conditional classes in the `index.js`)_.
+
+Migrating the styling from the `styles.css` to the `styles.js` will need some manual work (or a bit more for complex components). You'll have to convert the css to js, but luckily there are many online [css-to-js transformers](#https://transform.tools/css-to-js) that can help.
+
+Create a `styles.js` file next to the `theme.js`. (Don't forget the copyright block!)
```

Comment:
I would not mention the copyright block, who knows what coding standards people have in their codebases

Reply:
True, removed it, thanks.